src/test/java/org/openRealmOfStars/player/ship/ShipDesignTest2.java

```diff
@@ -0,0 +1,564 @@
+	    ShipDesign design = new ShipDesign(shiphull);
+	    assertEquals(NO_TROOPER_SHIP_POWER,design.getTotalTrooperPower());
+	    
+	    ShipComponent invasionModule = Mockito.mock(ShipComponent.class);
+	    Mockito.when(invasionModule.getType()).thenReturn(ShipComponentType.PLANETARY_INVASION_MODULE);
+	    Mockito.when(invasionModule.getDamage()).thenReturn(PLANETARY_INVASION_MODULE_DAMAGE);
+	    
+	    ShipComponent shield = Mockito.mock(ShipComponent.class);
+	    Mockito.when(shield.getType()).thenReturn(ShipComponentType.ARMOR);
+	    Mockito.when(shield.getDefenseValue()).thenReturn(ARMOR_VALUE);
+
+	    ShipComponent armor = Mockito.mock(ShipComponent.class);
+	    Mockito.when(armor.getType()).thenReturn(ShipComponentType.SHIELD);
+	    Mockito.when(armor.getDefenseValue()).thenReturn(SHIELD_VALUE);
+	    
+	    ShipComponent engine = Mockito.mock(ShipComponent.class);
+	    Mockito.when(engine.getType()).thenReturn(ShipComponentType.ENGINE);
+	    Mockito.when(engine.getEnergyResource()).thenReturn(ENGINE_VALUE);
+	    
+	    design.addComponent(invasionModule);
+	    design.addComponent(shield);
+	    design.addComponent(armor);
+	    design.addComponent(engine);
+	    
+	    assertEquals(TOTAL_POWER, design.getTotalTrooperPower());
+	    
+	    ShipHull shiphull2 = Mockito.mock(ShipHull.class);
+		Mockito.when(shiphull2.getSlotHull()).thenReturn(SLUT_HULL);
+		Mockito.when(shiphull2.getMaxSlot()).thenReturn(1);
+	    ShipDesign design2 = new ShipDesign(shiphull2);
+	    ShipComponent invasionModule2 = Mockito.mock(ShipComponent.class);
+	    Mockito.when(invasionModule2.getType()).thenReturn(ShipComponentType.PLANETARY_INVASION_MODULE);
+	    Mockito.when(invasionModule2.getDamage()).thenReturn(PLANETARY_INVASION_MODULE_DAMAGE);
+	    design2.addComponent(invasionModule2);
+	    assertEquals(NO_TROOPER_SHIP_POWER,design2.getTotalTrooperPower());
+	   
+	}
+	
+	/** 
+	 * getComponentListTest
+	 * purpose : Make sure getComponentList is executed
+	 * input : 
+	 * 				ShipComponent[] testResult ( engine,shield,armor)
+	 * expected :
+	 * 				ShipDesign.getComponentList() == testResult
+	 */
+	@Test
+	public void getComponentListTest(){
+		final int COMPONENT_NUM = 3;
+		ShipComponent[] testResult = new ShipComponent[COMPONENT_NUM];
+		
+		ShipHull shiphull = Mockito.mock(ShipHull.class);
+		ShipDesign design = new ShipDesign(shiphull);
+		
+		ShipComponent engine = Mockito.mock(ShipComponent.class);
+		ShipComponent shield = Mockito.mock(ShipComponent.class);
+	    ShipComponent armor = Mockito.mock(ShipComponent.class);
+	    
+	    testResult[0] = engine;
+	    testResult[1] = shield;
+	    testResult[2] = armor;
+	    
+	    design.addComponent(engine);
+	    design.addComponent(shield);
+	    design.addComponent(armor);
+	    
+	    assertEquals(testResult, design.getComponentList());
+	}
+	
+	/** 
+	 * changePriorityTest
+	 * purpose : Make sure changePriority is executed
+	 * input : 
+	 * 				UNVALID_INDEX is 3

```

Comment:
Just a small typo: UNVALID probably should be INVALID. There are couple of other position where UNVALID is mentioned.